### PR TITLE
[MODINV-955] Implement endpoint for updating ownership of Items

### PR DESCRIPTION
**Pull request by RomanChernetskyi**
### Purpose 
Implement endpoint for updating ownership of Items

### Approach
- Verify that target HoldingsRecord exists and linked to shared Instance
- Validate items to be linked to shared Instance
- Create Items on target tenant
- Delete items on source tenant

Jira: https://folio-org.atlassian.net/browse/MODINV-955

### CODE REVIEW

**Review by JavokhirAbdullayev — COMMENTED** [review]
@RomanChernetskyi please add a link to the ticket in jira

File: src/main/java/org/folio/inventory/resources/UpdateOwnershipApi.java [hunk 1]
```diff
@@ -125,7 +137,112 @@ private void processUpdateHoldingsOwnership(RoutingContext routingContext) {
...
+        itemsUpdateOwnership.getTargetTenantId());
+
+      consortiumService.getConsortiumConfiguration(context).toCompletionStage().toCompletableFuture()
+        .thenCompose(consortiumConfigurationOptional -> {
+          if (consortiumConfigurationOptional.isPresent()) {
+            Context targetTenantContext = constructContext(itemsUpdateOwnership.getTargetTenantId(), context.getToken(), context.getOkapiLocation());
+            return storage.getHoldingsRecordCollection(targetTenantContext)
+              .findById(itemsUpdateOwnership.getToHoldingsRecordId())
+              .thenCompose(holdingsRecord -> {
+                if (holdingsRecord != null) {
```
[the remark this answers is not in the document]
Comment (RomanChernetskyi) [reply src/main/java/org/folio/inventory/resources/UpdateOwnershipApi.java:164]: Thanks, done

[the remark this answers is not in the document]
Comment (JavokhirAbdullayev) [reply src/main/java/org/folio/inventory/resources/UpdateOwnershipApi.java:164]: looks better thank you Roman!

**Review by JavokhirAbdullayev — APPROVED** [review]

File: src/main/java/org/folio/inventory/resources/UpdateOwnershipApi.java [hunk 2]
```diff
@@ -125,7 +137,116 @@ private void processUpdateHoldingsOwnership(RoutingContext routingContext) {
...
+
+      if (validationError.isPresent()) {
+        unprocessableEntity(routingContext.response(), validationError.get());
+        return;
+      }
+      var itemsUpdateOwnership = updateOwnershipRequest.mapTo(ItemsUpdateOwnership.class);
+
+      List<NotUpdatedEntity> notUpdatedEntities = new ArrayList<>();
+
+      LOGGER.info("updateHoldingsOwnership:: Started updating ownership of item record: {}, to tenant: {}", itemsUpdateOwnership.getItemIds(),
```
Comment (KaterynaSenchenko) [inline src/main/java/org/folio/inventory/resources/UpdateOwnershipApi.java:154]: "updateHoldingsOwnership" - looks like log message should be corrected, and maybe LOG_UPDATE_ITEMS_OWNERSHIP constant used as in other places

File: src/main/java/org/folio/inventory/resources/UpdateOwnershipApi.java [hunk 3]
```diff
@@ -125,7 +137,116 @@ private void processUpdateHoldingsOwnership(RoutingContext routingContext) {
...
+                }
+              });
+          }
+          String notInConsortiaErrorMessage = format(TENANT_NOT_IN_CONSORTIA, context.getTenantId());
+          LOGGER.warn(format(LOG_UPDATE_HOLDINGS_OWNERSHIP, notInConsortiaErrorMessage));
+          return CompletableFuture.failedFuture(new BadRequestException(notInConsortiaErrorMessage));
+        })
+        .thenAccept(v -> respond(routingContext, notUpdatedEntities))
+        .exceptionally(throwable -> {
+          LOGGER.warn("updateHoldingsOwnership:: Error during update ownership of items {}, to tenant: {}",
```
Comment (KaterynaSenchenko) [inline src/main/java/org/folio/inventory/resources/UpdateOwnershipApi.java:180]: same here

File: src/main/java/org/folio/inventory/resources/UpdateOwnershipApi.java [hunk 4]
```diff
@@ -125,7 +137,116 @@ private void processUpdateHoldingsOwnership(RoutingContext routingContext) {
...
+            itemsUpdateOwnership.getItemIds(), itemsUpdateOwnership.getTargetTenantId(), throwable);
+          handleFailure(throwable, routingContext);
+          return null;
+        });
+
+    } catch (Exception e) {
+      LOGGER.warn("updateHoldingsOwnership:: Error during update ownership of items", e);
```
Comment (KaterynaSenchenko) [inline src/main/java/org/folio/inventory/resources/UpdateOwnershipApi.java:187]: and here

**Review by KaterynaSenchenko — APPROVED** [review]

File: src/main/java/org/folio/inventory/resources/UpdateOwnershipApi.java (hunk 2 above)
Comment (RomanChernetskyi) [reply src/main/java/org/folio/inventory/resources/UpdateOwnershipApi.java]: Done, Thanks

File: descriptors/ModuleDescriptor-template.json [hunk 5]
```diff
@@ -548,7 +548,8 @@
             "inventory-storage.items.item.post",
             "inventory-storage.items.item.delete",
             "inventory-storage.items.collection.get",
-            "inventory-storage.holdings.item.get"
+            "inventory-storage.holdings.item.get",
```
[the remark this answers is not in the document]
Comment (RomanChernetskyi) [reply descriptors/ModuleDescriptor-template.json:551]: Done, Thank you!

**Pull request merged**